applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp

```diff
@@ -224,97 +195,22 @@ void ImportData_ModelPart_Scalar(
 
     CoSimIO::ImportData(rInfo, DataBuffers::vector_doubles);
 
-    if (DataLoc == DataLocation::NodeHistorical) {
-        ImportDataSizeCheck(rModelPart.NumberOfNodes(), DataBuffers::vector_doubles.size());
-        std::size_t counter = 0;
-        for (auto& r_node : rModelPart.Nodes()) {
-            r_node.FastGetSolutionStepValue(rVariable) = DataBuffers::vector_doubles[counter++];
-        }
-
-    } else if (DataLoc == DataLocation::NodeNonHistorical) {
-        ImportDataSizeCheck(rModelPart.NumberOfNodes(), DataBuffers::vector_doubles.size());
-        std::size_t counter = 0;
-        for (auto& r_node : rModelPart.Nodes()) {
-            r_node.GetValue(rVariable) = DataBuffers::vector_doubles[counter++];
-        }
-
-    } else if (DataLoc == DataLocation::Element) {
-        ImportDataSizeCheck(rModelPart.NumberOfElements(), DataBuffers::vector_doubles.size());
-        std::size_t counter = 0;
-        for (auto& r_elem : rModelPart.Elements()) {
-            r_elem.GetValue(rVariable) = DataBuffers::vector_doubles[counter++];
-        }
-
-    } else if (DataLoc == DataLocation::Condition) {
-        ImportDataSizeCheck(rModelPart.NumberOfConditions(), DataBuffers::vector_doubles.size());
-        std::size_t counter = 0;
-        for (auto& r_cond : rModelPart.Conditions()) {
-            r_cond.GetValue(rVariable) = DataBuffers::vector_doubles[counter++];
-        }
-
-    } else if (DataLoc == DataLocation::ModelPart) {
-        ImportDataSizeCheck(1, DataBuffers::vector_doubles.size());
-        rModelPart[rVariable] = DataBuffers::vector_doubles[0];
-    }
+    AuxiliarModelPartUtilities Aux_Model_Part = AuxiliarModelPartUtilities(rModelPart);
+    Aux_Model_Part.SetScalarData<double>(rVariable, (Kratos::DataLocation)DataLoc, DataBuffers::vector_doubles);
 
     KRATOS_CATCH("")
 }
 
 void ExportData_ModelPart_Vector(
     CoSimIO::Info& rInfo,
-    const ModelPart& rModelPart,
+    ModelPart& rModelPart,
```

Comment:
```suggestion
    const ModelPart& rModelPart,
```

this should remain `const`, should be possible after making the `Get...` functions `const`

Reply:
I did this, but then there was an error while calling the constructor of the class AuxiliarModelPartUtilities(). 
But then I am able to solve this, by calling the constructor like this 
_**AuxiliarModelPartUtilities((ModelPart&) rModelPart)**_

I will commit it.

Reply:
ah I see
then leave it without the `const`, we shouldn't use the casts in this case

Reply:
Ohh okay ... fine .. I will remove the cast then ..